packages/node_modules/@webex/webex-core/test/unit/spec/interceptors/auth.js

```diff
@@ -136,136 +141,157 @@ describe('webex-core', () => {
+              uri: `${services.hydra}/ping`
+            });
+          });
+
+        it('resolves to true when the u2c service is specified via service',
+          () => {
+            services = {};
+
+            return interceptor.requiresCredentials({
+              service: 'u2c',
+              resource: 'something'
             })
-              .then((result) => assert.deepEqual(result, {
-                uri: 'https://not-a-service.sample.com/ping',
-                addAuthHeader: false,
-                headers: {} // doesn't add auth token for non-whitelisted domain
-              }))
-          ]));
+              .then((response) => assert.isTrue(response));
+          });
 
-          it('dont add header if `validatedomain` is false', () => {
-            webex.config.device.validateDomains = false;
+        it('resolves to false when the u2c limited service is used via uri',
+          () => interceptor.requiresCredentials({
+            uri: `${services.u2c}/limited`
+          })
+            .then((response) => assert.isFalse(response)));
+
+        it('resolves to true if the service exists in catalog via service',
+          () => interceptor.requiresCredentials({service: 'hydra'})
+            .then((response) => assert.isTrue(response)));
 
-            return interceptor.onRequest({
-              uri: 'https://not-a-service.wbx2.com/ping'
+        it('resolves to true if the service exists in catalog via uri',
+          () => interceptor.requiresCredentials({uri: services.hydra})
+            .then((response) => assert.isTrue(response)));
+
+        it('resolves to false if that `addAuthHeader` is set to false',
+          () => interceptor.requiresCredentials({
+            addAuthHeader: false,
+            service: 'unknown',
+            resource: 'ping'
+          })
+            .then((response) => assert.isFalse(response)));
+
+        it('resolves to false if `validateDomains` is set to false',
+          () => {
+            webex.internal.services.validateDomains = false;
+
+            return interceptor.requiresCredentials({
+              uri: 'https://allowed-uri.com/resource'
             })
-              .then((result) => assert.deepEqual(result, {
-                uri: 'https://not-a-service.wbx2.com/ping',
-                headers: {}
-              }));
+              .then((response) => assert.isFalse(response));
           });
+
+        it('resolves to true with an allowed domain uri',
+          () => interceptor.requiresCredentials({
+            uri: `https://${config.services.allowedDomains[0]}/resource`
+          })
+            .then((response) => assert.isTrue(response)));
+
+        it('resolves to false with a non-allowed uri',
+          () => interceptor.requiresCredentials({
+            uri: 'https://not-allowed/resource'
+          })
+            .then((response) => assert.isFalse(response)));
+
+        afterEach('remove services plugin', () => {
```

Comment:
Could you move this to immediately after the `beforeEach` hook for consistency with the other tests? (Trying to find an eslint rule like this jest one: https://github.com/jest-community/eslint-plugin-jest/blob/HEAD/docs/rules/prefer-hooks-on-top.md)